pkg/controller/core/queuedworkload_controller.go

```diff
@@ -142,12 +149,18 @@ func (r *QueuedWorkloadReconciler) Update(e event.UpdateEvent) bool {
 	case prevStatus == pending && status == pending:
 		if !r.queues.UpdateWorkload(oldWl, wl.DeepCopy()) {
 			log.V(2).Info("Queue for updated workload didn't exist; ignoring for now")
+			if err := r.cache.UpdateWorkloadStatus(context.Background(), wl, kueue.QueuedWorkloadPending, "Pending", "Queue for workload didn't exist"); err != nil {
+				log.V(2).Error(err, "Failed to update QueuedWorkload Status")
+			}
 		}
 
 	case prevStatus == pending && status == admitted:
 		r.queues.DeleteWorkload(oldWl)
 		if !r.cache.AddOrUpdateWorkload(wl.DeepCopy()) {
 			log.V(2).Info("ClusterQueue for workload didn't exist; ignored for now")
+			if err := r.cache.UpdateWorkloadStatus(context.Background(), wl, kueue.QueuedWorkloadPending, "Pending", "ClusterQueue for workload didn't exist"); err != nil {
```

Comment:
ditto, lets not set the condition here.

Reply:
Why ? will only land here IF the case is meet and the `AddOrUpdate` fails.